Shared/Realm.Sync.Shared/Permissions/IPermissionObject.cs

```diff
@@ -66,15 +66,20 @@ public interface IPermissionObject
         /// </listheader>
         /// </list>
         /// </remarks>
-        int? StatusCode { get; set; }
+        int? StatusCode { get; }
 
         /// <summary>
-        /// Gets or sets the status message.
+        /// Gets the status message.
         /// </summary>
         /// <remarks>
         /// Filled by the server after an object was processed with additional info
         /// explaining the status if necessary.
         /// </remarks>
-        string StatusMessage { get; set; }
+        string StatusMessage { get; }
+
+        /// <summary>
+        /// Gets the <see cref="ManagementObjectStatus"/> as set by the server.
+        /// </summary>
+        ManagementObjectStatus ObjectStatus { get; }
```

Comment:
I don't like the name `ObjectStatus` as you have to go look at the type to infer its meaning. Maybe `ManagementStatus`?